lighthouse-core/gather/driver.js

```diff
@@ -630,6 +630,60 @@ class Driver {
     return new LHElement(targetNode, this);
   }
 
+  /**
+   * @param {string} handle
+   * @param {{timeout: number}=} options,
+   * @return {Promise<string>}
+   */
+  async readIOStream(handle, options = {timeout: 5000}) {
```

Comment:
I was concerned about spending too much time making an unknown number of calls to `IO.read` so I added this timeout.